comp/core/workloadmeta/collectors/internal/process/process_collector.go

```diff
@@ -0,0 +1,148 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package process implements the local process collector for
+// Workloadmeta.
+package process
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"go.uber.org/fx"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/errors"
+	processwlm "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	collectorID   = "local-process-collector"
+	componentName = "workloadmeta-process"
+)
+
+type collector struct {
+	id      string
+	store   workloadmeta.Component
+	catalog workloadmeta.AgentType
+
+	processDiffCh <-chan *processwlm.ProcessCacheDiff
+}
+
+func collectorEventsFromProcessDiff(diff *processwlm.ProcessCacheDiff) []workloadmeta.CollectorEvent {
```

Comment:
[minor]
- For these private functions, let's put it towards the bottom of the file. 
- Lets add documentation on what this does
- Maybe a better name for this method would be `transform`. The args and return type will implicitly tell the user what this transforms from `ProcessCacheDiff` to `CollectorEvent`

Reply:
done.